src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/transformer/ApplicationsTransformer.kt

```diff
@@ -58,9 +59,25 @@ class ApplicationsTransformer(
     submittedAt = jpa.submittedAt.toInstant()
   )
 
-  private fun getStatus(entity: ApplicationEntity) = when {
-    entity.getLatestAssessment()?.clarificationNotes?.any { it.response == null } == true -> ApplicationStatus.requestedFurtherInformation
-    entity.submittedAt !== null -> ApplicationStatus.submitted
-    else -> ApplicationStatus.inProgress
+  private fun getStatus(entity: ApplicationEntity): ApplicationStatus {
+    val latestAssessment = entity.getLatestAssessment()
+
+    if (entity is ApprovedPremisesApplicationEntity) {
+      return when {
+        latestAssessment?.submittedAt != null && latestAssessment.decision == AssessmentDecision.REJECTED -> ApplicationStatus.rejected
+        latestAssessment?.submittedAt != null && latestAssessment.decision == AssessmentDecision.ACCEPTED && entity.getLatestPlacementRequest() == null -> ApplicationStatus.pending
+        latestAssessment?.submittedAt != null && latestAssessment.decision == AssessmentDecision.ACCEPTED && entity.getLatestPlacementRequest()!!.booking == null -> ApplicationStatus.awaitingPlacement
```

Comment:
Could we have a `getBooking()` function on the Application? I imagine it'd be useful elswhere too.